src/betree.h

```diff
@@ -0,0 +1,139 @@
+#ifndef BETREE_H__
+#define BETREE_H__
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+struct pred {
+    unsigned int variable_id;
```

Comment:
This is somewhat subjective but I tend to rarely if ever use regular int types in C code. The reason being that their size can vary between platform so I prefer using `size_t`, `ssize_t` and the int types in `stdint.h` which usually better reflect the limits I want to impose on the domain of the variables. `size_t` is particularly useful as it's defined as being big enough to index every possible array in a array (ie. uint32_t on 32 bit systems and uint64_t on 64 bit systems).

I also frequently wrap my int types in `typedef` to make sure that various parts of my code agree on types. As an example you have various ids in these structs so creating something along the lines of `typedef id_t uint64_t;` would ensure a consistent type across all ids.

Note that ints are the worst part of C (tons of undefined behaviour) so being careful with them can have long term benefits :)